packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -528,9 +454,146 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
 
             IController(controller).withdrawUniPositionToken(data.vaultId);
 
-            _closeUniLp(data);
+            (uint256 wPowerPerpAmountInLp, ) = ControllerHelperUtil.closeUniLp(
+                nonfungiblePositionManager,
+                ControllerHelperDataType.closeUniLpParams({
+                    tokenId: data.tokenId,
+                    liquidity: data.liquidity,
+                    liquidityPercentage: data.liquidityPercentage,
+                    amount0Min: data.amount0Min,
+                    amount1Min: data.amount1Min
+                }),
+                isWethToken0
+            );
+
+            // close short position using amounts collected from closing LP, withdraw collateralToWithdraw + deposited collateralToFlashloan
+            _closeShortWithAmountsFromLp(
+                data.vaultId,
+                wPowerPerpAmountInLp,
+                data.wPowerPerpAmountToBurn,
+                data.collateralToWithdraw.add(data.collateralToFlashloan),
+                data.limitPriceEthPerPowerPerp
+            );
+            ControllerHelperUtil.checkPartialLpClose(
+                controller,
+                nonfungiblePositionManager,
+                data.vaultId,
+                data.tokenId,
+                data.liquidityPercentage
+            );
```

Comment:
But this will close fully or partially vault debt using vault NFT, so we will always get WETH from `closeUniLp()` and `_closeShortWithAmountsFromLp()`. 

Or is there a case where fee will be sent from user as ETH ?